src/anamoly_detection/anomaly_engine.py

```diff
@@ -0,0 +1,76 @@
+from scapy.all import Packet
+from src.anamoly_detection.equation_parser import parse_equation
+from src.database.models import AnomalyEquations
+from src.anamoly_detection.frequency_signatures.traffic.traffic_layer_frequency_signature \
+    import TrafficLayerFrequencySignature
+
+
+class AnomalyEngine:
+    """
+    Engine object used to manage anomaly signatures
+    """
+    def __init__(self, connection, frequency_signatures=[], traffic_signatures=[]):
+        """
+        Creates an engine object using the given connection, defaulting to no signatures present
+        :param connection: DB connection used to retrieve limit information
+        :param frequency_signatures: Frequency based signatures (Time based limits)
+        :param traffic_signatures: Traffic based signatures (Traffic type based limits)
+        """
+        # Save connection and create a session if necessary
+        self.connection = connection
+        if not self.connection.session:
+            self.connection.create_session()
```

Comment:
Anomaly engine should leverage the existing connection in __init__.py instead of creating its own.

Reply:
It is leveraging that, only makes a connection when there is none